openc3/python/openc3/microservices/interface_microservice.py

```diff
@@ -413,7 +428,11 @@ def graceful_kill(self):
         time.sleep(0.001)  # Allow other threads to run
 
     def run(self):
-        for topic, msg_id, msg_hash, redis in RouterTopic.receive_telemetry(self.router, scope=self.scope):
+        generator = RouterTopic.receive_telemetry(self.router, scope=self.scope)
+        topic, msg_id, msg_hash, _redis = next(generator)
```

Comment:
This being inside of a `for` loop meant that the old `return` statements would break the loop and cause the thread to exit which was detected by the operator and then restarted and immediately would reconnect. This was the source of the bug.